### PR TITLE
{{🚀[Feat]: Added Feature to upload files locally}}

### PR DESCRIPTION
**Pull request by jainaayush01**
Closes Issue: #57

### CODE REVIEW

**Comment by kothariji** [comment]
Hi @jainaayush01, Thank you for the PR.
But actually, we are working on Refactoring and implementing redux. You can see PR #99, so once it is merged, then we will work on this PR for sure.

Please bare with us 🙌 😊

**Comment by kothariji** [comment]
Hi @jainaayush01, we are done with the refactoring of code, so I would request you to remove the merge conflicts so we can preview the PR, and merge it once done. 🙌

**Comment by jainaayush01** [comment]
Done.

**Comment by kothariji** [comment]
Also, there is a compilation error 
image

Please check once running code in local 🙌

File: frontend/syntaxmeets/src/components/SyntaxEditor/SyntaxEditor.js [hunk 1]
```diff
@@ -85,6 +86,89 @@ const SyntaxEditor = (props) => {
     setPopup(true);
   };
 
+  const uploadFile = () => {
+    document.querySelector("#upload").click();
+  }
+
+  const handleInputChange = (newInput) => {
+    setCodeInput(newInput);
+  };
```
Comment (kothariji) [inline frontend/syntaxmeets/src/components/SyntaxEditor/SyntaxEditor.js:95]: Also please remove that, as we are handling this part in separate component `INPUT`

**Review by kothariji — CHANGES_REQUESTED** [review]
Please do the requested changes

**Comment by jainaayush01** [comment]
> Also, there is a compilation error
> image
> 
> Please check once running code in local 🙌

my bad i should have checked!! sorry for that

**Comment by kothariji** [comment]
> I have copied the SyntaxEditor.js file from the main branch here and have added my changes there. Still, I am seeing 2 more file changes. please let me know if some more changes are to be made or should I just make a new PR with an updated codebase.

Preview Link - https://syntaxmeets-6urefc9zf-kothariji.vercel.app/
### Hey @jainaayush01, the PR looks amazing, things are working great 🙌 
⚠️ but there are a few bugs

 - Even if we have explicitly mentionedthat we will accept only  **.c, .cpp, .java, .js, .ts, .clj, .cljs, .cs, .cbl, .cob, .cpy, .erl,**, but then also we are able to upload .jpeg, .exe etc. So can we validate the file extension in handleFileChange function, and if its not valid we can show an alert that, please upload the supported files only.
 
 - Another issue is that currently there are no such restrictions on the size of the file, but if you'll try uploading big files, the socket will be disconnected.  So it can be great if we can restrict the file size for now to max **'10 kb'**.
 
 Please let me know for the same 🙌

**Comment by jainaayush01** [comment]
I have fixed the above mentioned bugs if still there's something missing out please let me know.
🙌

**Review by kothariji — APPROVED** [review]
@jainaayush01 tremendous work 🚀 
Thank you for co-operating and doing the suggested changes.  We really appreciate your contribution 🙌 

Keep Contributing 🔥 🙏

**Pull request merged**